lib/chai/core/assertions.js

```diff
@@ -303,11 +303,12 @@ module.exports = function (chai, _) {
 
     if (msg) flag(this, 'message', msg);
     var obj = flag(this, 'object')
+      , objType = _.type(obj).toLowerCase()
       , isDeep = flag(this, 'deep')
       , descriptor = isDeep ? 'deep ' : '';
 
     // This block is for asserting a subset of properties in an object.
-    if (_.type(obj) === 'object') {
+    if (objType === 'object') {
```

Comment:
I added the `.toLowerCase` anytime that excluding it would mean that we'd have to perform additional comparisons as a result. For example, on line 346 we'd have to perform a second comparison for "String" (to handle the object version) in addition to the current one for "string".